addons/website/static/src/components/dialog/dialog.js

```diff
@@ -124,57 +116,406 @@ export class AddPageDialog extends Component {
+            for (const cssLinkEl of await this.props.getCssLinkEls()) {
+                const preloadLinkEl = document.createElement("link");
+                preloadLinkEl.setAttribute("rel", "preload");
+                preloadLinkEl.setAttribute("href", cssLinkEl.getAttribute("href"));
+                preloadLinkEl.setAttribute("as", "style");
+                iframeEl.contentDocument.head.appendChild(preloadLinkEl);
+                iframeEl.contentDocument.head.appendChild(cssLinkEl.cloneNode(true));
+            }
+            // Adjust styles.
+            const styleEl = document.createElement("style");
+            // Does not work with fit-content in Firefox.
+            const carouselHeight = isFirefox ? '450px' : 'fit-content';
+            // Prevent successive resizes.
+            const fullHeight = getComputedStyle(document.querySelector(".o_action_manager")).height;
+            const halfHeight = `${Math.round(parseInt(fullHeight) / 2)}px`;
+            const css = `
+                #wrapwrap {
+                    overflow: hidden;
+                    padding-right: 0px;
+                    padding-left: 0px;
+                }
+                section {
+                    ${" " || "Avoid the zoom's missing pixel."}
+                    transform: scale(101%);
+                }
+                section[data-snippet="s_carousel"],
+                section[data-snippet="s_quotes_carousel"] {
+                    height: ${carouselHeight} !important;
+                }
+                section.o_half_screen_height {
+                    min-height: ${halfHeight} !important;
+                }
+                section.o_full_screen_height {
+                    min-height: ${fullHeight} !important;
+                }
+                section[data-snippet="s_three_columns"] .figure-img[style*="height:50vh"] {
+                    ${" " || "In Travel theme."}
+                    height: 170px !important;
+                }
+                .o_we_shape {
+                    ${" " || "Avoid the zoom's missing pixel."}
+                    transform: scale(101%);
+                }
+            `;
+            const cssText = document.createTextNode(css);
+            styleEl.appendChild(cssText);
+            iframeEl.contentDocument.head.appendChild(styleEl);
+            // Put blocks.
+            // To preserve styles, the whole #wrapwrap > main > #wrap
+            // nesting must be reproduced.
+            const mainEl = document.createElement("main");
+            const wrapwrapEl = document.createElement("div");
+            wrapwrapEl.id = "wrapwrap";
+            wrapwrapEl.appendChild(mainEl);
+            iframeEl.contentDocument.body.appendChild(wrapwrapEl);
+            const templateDocument = new DOMParser().parseFromString(this.props.template.template, "text/html");
+            const wrapEl = templateDocument.getElementById("wrap");
+            // Clean-up data-oe- attributes.
```

Comment:
You shouldn't have branding here, there is surely a way to bypass the branding in your code.
Check `inherit_branding`